completion/kubectx.zsh

```diff
@@ -2,11 +2,17 @@
 
 local KUBECTX="${HOME}/.kube/kubectx"
 PREV=""
+
+local all_contexts="$(kubectl config get-contexts --output='name')"
 if [ -f "$KUBECTX" ]; then
     # show '-' only if there's a saved previous context
     local PREV=$(cat "${KUBECTX}")
-    _arguments "1: :(-
-        $(kubectl config get-contexts --output='name'))"
+
+    _arguments \
+      "-d:*: :(${all_contexts})" \
+      "(- *): :(- ${all_contexts})"
```

Comment:
thanks a lot for this PR!
Can you explain this syntax and how it works real quick for me?

Reply:
Hi, of course.

`"-d:*: :(${all_contexts})"`
- -d -> The kubectx subcommand.
- '*' -> This describes multiple arguments. All contexts matching the pattern (`*`, all) will be completed.
- ' ' -> An empty description.
- (${all_contexts}) -> An action describing all words to be completed.

`"(- *): :(- ${all_contexts})"`
- (- *) -> If one of the contexts has already been selected (and is on the command line), it will not complete other options anymore. (i.e this will avoid `kubectx ctx1 ctx2`)
- ' ' -> An empty description.
- (- ${all_contexts}) -> An action describing all words to be completed, including the current one (`-`).

Reply:
Why does the description have a space character to be “empty”?

Reply:
AFAIK the spec needs this space to know what `optspec` I'm using (`optname:*pattern:message:action` and no `optname:message:action`). I can add a description for them if you want. WDYT?

Reply:
Another relevant point is I'm not completing with all the current subcommands like `-d`, `-h` and `-c`. The subcommand `-d` will only be completed if the user enters the `-d` option (typing).

Reply:
Yeah, makes sense. Thanks.
I might find you again when there's an issue with this file. 😄

Reply:
Ok 😄